packages/flutter/lib/src/painting/continuous_rectangle_border.dart

```diff
@@ -3,58 +3,251 @@
 // found in the LICENSE file.
 
 import 'dart:math' as math;
+import 'dart:ui' as ui;
 
 import 'basic_types.dart';
-import 'border_radius.dart';
 import 'borders.dart';
 import 'edge_insets.dart';
 
-/// A rectangular border with smooth continuous transitions between the straight
-/// sides and the rounded corners.
+/// A rectangle border with continuous corners.
+///
+/// A shape similar to a rounded rectangle, but with a smoother transition from
+/// the sides to the rounded corners.
+///
+/// The rendered shape roughly approximates that of a superellipse. In this
+/// shape, the curvature of each 90º corner around the length of the arc is
+/// approximately a gaussian curve instead of a step function as with a
+/// traditional quarter circle round. The rendered rectangle is roughly a
+/// superellipse with an n value of 5.
+///
+/// In an attempt to keep the shape of the rectangle the same regardless of its
+/// dimension (and to avoid clipping of the shape), the radius will
+/// automatically be lessened if its width or height is less than ~3x the
+/// declared radius. The new resulting radius will always be maximal in respect
+/// to the dimensions of the given rectangle.
+///
+/// To achieve the corner smoothness with this shape, ~50% more pixels in each
+/// dimensions are used to render each corner. As such, the ~3 represents twice
```

Comment:
This still reads like too much math in the sentence. Just "Therefore, with 2 corners per side, the height or width needs to be minimally ~1.5 * 2 = ~3 times the length of the radius to accommodate the smooth corner transitions."